tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot

```diff
@@ -155,20 +155,43 @@ Clean Up User Notebook
   [Arguments]  ${admin_username}  ${username}
 
   # Verify that ${admin_username}  is connected to the cluster
-  ${oc_whoami}=  Run   oc whoami
+  ${oc_whoami} =  Run   oc whoami
   IF    '${oc_whoami}' == '${admin_username}'
       # We import the library here so it's loaded only when we are connected to the cluster
       # Having the usual "Library OpenShiftCLI" in the header raises an error when loading the file
       # if there is not any connection opened
       Import Library    OpenShiftCLI
 
       # Verify that the jupyter notebook pod is running
-      ${notebook_pod_name}=   Get User Notebook Pod Name  ${username}
+      ${notebook_pod_name} =   Get User Notebook Pod Name  ${username}
       Search Pods    ${notebook_pod_name}  namespace=rhods-notebooks
 
       # Delete all files and folders in /opt/app-root/src/  (excluding hidden files/folders)
       # Note: rm -fr /opt/app-root/src/ or rm -fr /opt/app-root/src/* didn't work properly so we ended up using find
-      ${output}=  Run   oc exec ${notebook_pod_name} -n rhods-notebooks -- find /opt/app-root/src/ -not -path '*/\.*' -not -path '/opt/app-root/src/' -exec rm -rv {} +
+      ${output} =  Run   oc exec ${notebook_pod_name} -n rhods-notebooks -- find /opt/app-root/src/ -not -path '*/\.*' -not -path '/opt/app-root/src/' -exec rm -rv {} +
+      Log  ${output}
+  ELSE
+      Fail  msg=This command requires ${admin_username} to be connected to the cluster (oc login ...)
+  END
+
+Delete Folder In User Notebook
+  [Documentation]  Delete recursively the folder  /opt/app-root/src/${folder} in ${username}'s notebook PVC.
+...   Note: this command requires ${admin_username}  to be logged to the cluster (oc login ...) and to have the user's notebook pod running (e.g. jupyterhub-nb-ldap-2duser1)
+  [Arguments]  ${admin_username}  ${username}  ${folder}
+
+  # Verify that ${admin_username}  is connected to the cluster
+  ${oc_whoami} =  Run   oc whoami
+  IF    '${oc_whoami}' == '${admin_username}'
+      # We import the library here so it's loaded only when we are connected to the cluster
+      # Having the usual "Library OpenShiftCLI" in the header raises an error when loading the file
+      # if there is not any connection opened
+      Import Library    OpenShiftCLI
+
+      # Verify that the jupyter notebook pod is running
+      ${notebook_pod_name} =   Get User Notebook Pod Name  ${username}
+      Search Pods    ${notebook_pod_name}  namespace=rhods-notebooks
```

Comment:
Isn't this mostly a repetition of `Clean Up User Notebook`? Did you mean to remove the other keyword?

Reply:
This one removes an specific folder that you pass as a parameter. I think both can be useful

Reply:
Can't they be merged into one keyword? perhaps by setting the default value for `${folder}` to empty

Reply:
Personally I think it's more clear like this, one is to clear everything (except hidden files)  and the other to delete an specific folder. Maybe the fist one could have a better name (Empty User Notebook?)

`Clean Up User Notebook  ${OCP_ADMIN_USER.USERNAME}  ${TEST_USER.USERNAME}`

`Delete Folder In User Notebook  ${OCP_ADMIN_USER.USERNAME}  ${TEST_USER.USERNAME}  foldername`

Reply:
I personally prefer to keep code repetition to a minimum, but at this point it's just a question of personal preference. I think it works like this as well, I just wanted to check if you *really* wanted to have those two separate keyword or not.

Apart from this, lgtm

Reply:
resolving for now, but we can discuss in our call later.